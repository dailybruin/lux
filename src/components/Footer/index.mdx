---
name: Footer
route: /footer
---

import { Playground, PropsTable } from 'docz'
import Footer from '.'

# Footer

Every web page needs a footer.

<PropsTable of={Footer} />

<Playground>
  <Footer
    githubLink={'https://dailybruin.com/'}
    developers={['Nathan Smith', 'Michael Zhang', 'Dustin Newman']}
  />
</Playground>
