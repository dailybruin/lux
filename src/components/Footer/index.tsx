import * as React from 'react';
import { css } from 'react-emotion';
import Heart from './Heart';
import toSentence from '../../utils/toSentence';

enum License {
  MIT = 'MIT License',
  AGPL = 'GNU AGPL',
  Copyright = 'Copyright',
}

/**
 * Footer properties.
 */
interface FooterProps {
  /** The name of the github project. The `https://github.com/dailybruin/` part is already given. */
  githubName: string;
  /** The name of the license of the project. */
  license: License;
  /** A list of the developers who created the site. */
  developers: string | Array<string>;
  /** The year that the story was published, e.g., 2018. */
  copyrightYear: number;
}

/** A footer to go at the bottom of every page. */
class Footer extends React.Component<FooterProps> {
  static defaultProps = {
    license: License.Copyright,
  };

  render() {
    return (
      <footer
        className={css`
          display: flex;
          flex-direction: column;
          justify-content: center;
          margin-bottom: 1rem;
          text-align: center;
        `}
      >
        <div>
          Content copyright © {this.props.copyrightYear} Daily Bruin. Site code
          available on{' '}
          <a href={`https://github.com/dailybruin/${this.props.githubName}`}>
            GitHub
          </a>
          {this.props.license === License.Copyright
            ? '.'
            : ` and available under the ${this.props.license}.`}
        </div>
        <div>
          Built with Suzy’s <Heart /> in Kerckhoff 118 by{' '}
          {typeof this.props.developers === 'string'
            ? this.props.developers
            : toSentence(this.props.developers)}
          .
        </div>
      </footer>
    );
  }
}

export default Footer;
