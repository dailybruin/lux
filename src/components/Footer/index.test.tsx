import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import Footer from '.';

import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Footer', () => {
  afterEach(cleanup);

  it('Matches its snapshot', () => {
    const { container } = render(
      <Footer
        githubName="lux"
        developers={['Nathan Smith']}
        copyrightYear={2018}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
