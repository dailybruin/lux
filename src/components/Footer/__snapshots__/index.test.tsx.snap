// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Matches its snapshot 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
  text-align: center;
}

.emotion-0:hover {
  color: red;
  cursor: pointer;
}

<footer
  class="emotion-1"
>
  <div>
    Content copyright © 
    2018
     Daily Bruin. Site code available on
     
    <a
      href="https://github.com/dailybruin/lux"
    >
      GitHub
    </a>
    .
  </div>
  <div>
    Built with Suzy’s 
    <span
      class="emotion-0"
    >
      ♥
    </span>
     in Kerckhoff 118 by
     
    Nathan Smith
    .
  </div>
</footer>
`;
